src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            prev = null;
+            this.item = item;
+            next = null;
+        }
+    }
+
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value);
+        if (head == null) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddMethod(index);
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            newNode.next.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            newNode.prev = tail;
+            newNode.prev.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> previousToNew = findNodeByIndex(index - 1);
+            newNode.next = previousToNew.next;
+            newNode.prev = previousToNew;
+            previousToNew.next = newNode;
+            newNode.next.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        if (findNodeByIndex(index) == null) {
+            return null;
+        }
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexValidation(index);
+        Node<T> necessaryNode = findNodeByIndex(index);
+        T item = necessaryNode.item;
```

Comment:
```suggestion
        T oldValue = necessaryNode.item;
```
it will be more clear